util_curl.go

```diff
@@ -0,0 +1,77 @@
+package resty
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+
+	"net/url"
+	"strings"
+
+	"github.com/go-resty/resty/v2/shellescape"
+)
+
+func buildCurlRequest(req *http.Request, httpCookiejar http.CookieJar) (curl string) {
+	// 1. Generate curl raw headers
+	curl = "curl -X " + req.Method + " "
+	// req.Host + req.URL.Path + "?" + req.URL.RawQuery + " " + req.Proto + " "
+	headers := dumpCurlHeaders(req)
+	for _, kv := range *headers {
+		curl += `-H ` + shellescape.Quote(kv[0]+": "+kv[1]) + ` `
+	}
+
+	// 2. Generate curl cookies
+	if cookieJar, ok := httpCookiejar.(*cookiejar.Jar); ok{
+		cookies := cookieJar.Cookies(req.URL)
+		if len(cookies) > 0 {
+			curl += ` -H ` + shellescape.Quote(dumpCurlCookies(cookies)) + " "
+		}
+	}
+
+
+	// 3. Generate curl body
+	if req.Body != nil {
```

Comment:
@jeevatkm 
Deleting  middleware will cause GenerateCurlCommand to be unable to get the body.

 Because resty will delete the body after executing the request.
 
I think the best time to record the body is when the middleware request occurs. If it is before the middleware, parameters such as url will be lost. If it is after the middleware, the body parameters will be lost.